File: templates/workspace-dialect/Hello.java
package forge.app;

import forge.Op;
import forge.OpContext;
import forge.Value;

public final class Hello {

    public record Greeting(String message) {}

    @Op("hello")
    public static Greeting hello(OpContext ctx, Value input) {
        String name = input.get("name", "world");
        return new Greeting("hello, " + name + ", from {{domain}}");
    }
}

File: templates/workspace-dialect/hello.rs
use forge::{OpContext, Value};
use serde::Serialize;

#[derive(Serialize)]
pub struct Greeting {
    message: String,
}

pub fn hello(ctx: &OpContext, input: &Value) -> Greeting {
    let name: String = input.get_str("name", "world");
    Greeting {
        message: format!("hello, {}, from {{domain}}", name),
    }
}
